### PR TITLE
bugfix/#6281

### PR DESCRIPTION
**Pull request by KKatiaa**
After clicking on friend's request card user is not redirected to user's page [#6281](https://github.com/ita-social-projects/GreenCity/issues/6281)
User can not accept friend request on User's page because system no not redirects to User's page [#6282](https://github.com/ita-social-projects/GreenCity/issues/6282)
Button 'Add friend' do not appear on User's page [#6283](https://github.com/ita-social-projects/GreenCity/issues/6283)
Message on Request page do not responds to requirements [#6284](https://github.com/ita-social-projects/GreenCity/issues/6284)

### CODE REVIEW

File: src/app/main/component/user/components/profile/users-friends/friend-dashboard/friend-profile-page/friend-profile-page.component.ts [hunk 1]
```diff
@@ -30,8 +35,11 @@ export class FriendProfilePageComponent implements OnInit, OnDestroy {
     this.translate.setDefaultLang(this.localStorageService.getCurrentLanguage());
     this.bindLang();
     this.userId = this.route.snapshot.params.userId;
+    this.isRequest = this.router.url?.includes('requests');
+    console.log(this.isRequest, 'isRequest');
```
Comment (hnativlyubomyr) [inline src/app/main/component/user/components/profile/users-friends/friend-dashboard/friend-profile-page/friend-profile-page.component.ts]: remove console.log

File: src/app/main/component/user/components/profile/users-friends/friend-dashboard/friend-profile-page/friend-profile-page.component.ts [hunk 2]
```diff
@@ -56,6 +64,40 @@ export class FriendProfilePageComponent implements OnInit, OnDestroy {
       });
   }
 
+  private deleteFriendsFromList(id, array) {
```
Comment (hnativlyubomyr) [inline src/app/main/component/user/components/profile/users-friends/friend-dashboard/friend-profile-page/friend-profile-page.component.ts]: add papameter types

File: src/app/main/component/user/components/profile/users-friends/friend-dashboard/friend-profile-page/friend-profile-page.component.ts [hunk 3]
```diff
@@ -56,6 +64,40 @@ export class FriendProfilePageComponent implements OnInit, OnDestroy {
...
+    const indexSuggestion = array.findIndex((item) => item.id === id);
+    array.splice(indexSuggestion, 1);
```
Comment (hnativlyubomyr) [inline src/app/main/component/user/components/profile/users-friends/friend-dashboard/friend-profile-page/friend-profile-page.component.ts]: the index here can be -1
I would return new array here:
return array.filter((item) => item.id !== id)

File: src/app/main/component/user/components/profile/users-friends/friend-dashboard/friend-profile-page/friend-profile-page.component.ts [hunk 4]
```diff
@@ -56,6 +64,40 @@ export class FriendProfilePageComponent implements OnInit, OnDestroy {
...
+  }
+
+  public accept(id: number) {
```
Comment (hnativlyubomyr) [inline src/app/main/component/user/components/profile/users-friends/friend-dashboard/friend-profile-page/friend-profile-page.component.ts]: add void result type

File: src/app/main/component/user/components/profile/users-friends/friend-dashboard/friend-profile-page/friend-profile-page.component.ts [hunk 5]
```diff
@@ -56,6 +64,40 @@ export class FriendProfilePageComponent implements OnInit, OnDestroy {
...
+    this.userFriendsService
+      .acceptRequest(id)
+      .pipe(takeUntil(this.destroy$))
+      .subscribe(() => {
+        this.deleteFriendsFromList(id, this.requests);
+      });
+    this.showButtons = false;
+  }
+
+  public decline(id: number) {
```
Comment (hnativlyubomyr) [inline src/app/main/component/user/components/profile/users-friends/friend-dashboard/friend-profile-page/friend-profile-page.component.ts]: the same

File: src/app/main/component/user/components/profile/users-friends/friend-dashboard/friend-profile-page/friend-profile-page.component.ts [hunk 6]
```diff
@@ -56,6 +64,40 @@ export class FriendProfilePageComponent implements OnInit, OnDestroy {
...
+    this.userFriendsService
+      .declineRequest(id)
+      .pipe(takeUntil(this.destroy$))
+      .subscribe(() => {
+        this.deleteFriendsFromList(id, this.requests);
+      });
+    this.showButtons = false;
+  }
+
+  private getRequests() {
```
Comment (hnativlyubomyr) [inline src/app/main/component/user/components/profile/users-friends/friend-dashboard/friend-profile-page/friend-profile-page.component.ts]: the same

File: src/app/main/component/user/components/profile/users-friends/friend-dashboard/friend-profile-page/friend-profile-page.component.html [hunk 7]
```diff
@@ -1,6 +1,11 @@
 <div class="profile-wrapper" *ngIf="userInfo">
   <div class="side-bar">
     <app-profile-header class="app-profile-header" [userInfo]="userInfo" [progress]="progress"></app-profile-header>
+    <hr *ngIf="isRequest && showButtons" />
+    <div class="friend-request" *ngIf="isRequest && showButtons">
+      <button class="secondary-global-button m-btn" (click)="accept(userId)">Accept</button>
```
Comment (hnativlyubomyr) [inline src/app/main/component/user/components/profile/users-friends/friend-dashboard/friend-profile-page/friend-profile-page.component.html]: Are translations needed here?

File: src/app/main/component/user/components/profile/users-friends/friend-dashboard/friend-profile-page/friend-profile-page.component.ts (hunk 1 above)
Comment (KKatiaa) [reply src/app/main/component/user/components/profile/users-friends/friend-dashboard/friend-profile-page/friend-profile-page.component.ts]: removed console.log

File: src/app/main/component/user/components/profile/users-friends/friend-dashboard/friend-profile-page/friend-profile-page.component.ts (hunk 2 above)
Comment (KKatiaa) [reply src/app/main/component/user/components/profile/users-friends/friend-dashboard/friend-profile-page/friend-profile-page.component.ts]: added papameter types

**Review by hnativlyubomyr — APPROVED** [review]

**Review by NadiiaStoiko — APPROVED** [review]

**Review by mdashko — APPROVED** [review]

**Review by sobkiv — APPROVED** [review]

**Review by Tadei-Mykola — APPROVED** [review]

**Pull request merged**